src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -408,12 +400,8 @@ private Optional<TaskNotificationType> recoverReplicateInRevealPhase(Task task,
         String chainTaskId = task.getChainTaskId();
         String walletAddress = replicate.getWalletAddress();
 
-        if (replicate.getLastRelevantStatus().isEmpty()) {
-            return Optional.empty();
-        }
-
-        boolean isInStatusContributed = replicate.getLastRelevantStatus().get().equals(ReplicateStatus.CONTRIBUTED);
-        boolean didReplicateStartRevealing = replicate.getLastRelevantStatus().get().equals(ReplicateStatus.REVEALING);
+        boolean isInStatusContributed = replicate.getLastRelevantStatus().equals(ReplicateStatus.CONTRIBUTED);
+        boolean didReplicateStartRevealing = replicate.getLastRelevantStatus().equals(ReplicateStatus.REVEALING);
```

Comment:
After our discussion, maybe ?
```suggestion
        boolean isInStatusContributed = replicate.getLastRelevantStatus() == ReplicateStatus.CONTRIBUTED;
        boolean didReplicateStartRevealing = replicate.getLastRelevantStatus() == ReplicateStatus.REVEALING;
```

Reply:
Should now be clean everywhere :)
https://github.com/iExecBlockchainComputing/iexec-core/pull/524/commits/7e35e70e9bf0d32693a888ee7d39e5e5596d8ed0